snippets/csharp/System.Collections.Generic/IEqualityComparerT/Overview/program.cs

```diff
@@ -2,81 +2,72 @@
 using System;
 using System.Collections.Generic;
 
-class Example
+static class Example
 {
-   static void Main()
-   {
-      BoxEqualityComparer boxEqC = new BoxEqualityComparer();
-
-      var boxes = new Dictionary<Box, string>(boxEqC);
-
-      var redBox = new Box(4, 3, 4);
-      AddBox(boxes, redBox, "red");
+    static void Main()
+    {
+        BoxEqualityComparer comparer = new();
 
-      var blueBox = new Box(4, 3, 4);
-      AddBox(boxes, blueBox, "blue");
+        Dictionary<Box, string> boxes = new(comparer);
 
-      var greenBox = new Box(3, 4, 3);
-      AddBox(boxes, greenBox, "green");
-      Console.WriteLine();
+        AddBox(new Box(4, 3, 4), "red");
+        AddBox(new Box(4, 3, 4), "blue");
+        AddBox(new Box(3, 4, 3), "green");
 
-      Console.WriteLine("The dictionary contains {0} Box objects.",
-                        boxes.Count);
-   }
+        Console.WriteLine($"The dictionary contains {boxes.Count} Box objects.");
 
-   private static void AddBox(Dictionary<Box, String> dict, Box box, String name)
-   {
-      try {
-         dict.Add(box, name);
-      }
-      catch (ArgumentException e) {
-         Console.WriteLine("Unable to add {0}: {1}", box, e.Message);
-      }
-   }
+        void AddBox(Box box, string name)
+        {
+            try
+            {
+                boxes.Add(box, name);
+            }
+            catch (ArgumentException e)
+            {
+                Console.WriteLine($"Unable to add {box}: {e.Message}");
+            }
+        }
+    }
 }
 
-public class Box
+class Box
 {
-    public Box(int h,  int l, int w)
-    {
-        this.Height = h;
-        this.Length = l;
-        this.Width = w;
-    }
+    public int Height { get; }
+    public int Length { get; }
+    public int Width { get; }
```

Comment:
I'd consider `init` accessors and a parameterless ctor here. What do you think?

```suggestion
    public int Height { get; init; }
    public int Length { get; init; }
    public int Width { get; init; }
    
    public Box() {}
```

Reply:
An explicit `public Box() { }` wouldn't be needed.

Reply:
> An explicit `public Box() { }` wouldn't be needed.

It is, unless the constructor on line 39 is removed.

Reply:
That works!

I originally wrote this with a record and primary constructor, but backed that out because records implement hashing and equality, making the comparer in the example obsolete. Maybe we could do better here by having an example for which the comparer does something more interesting and realistic.

Reply:
> It is, unless the constructor on line 39 is removed.

Oh, yeah. I assumed the ctor on L39 would be removed.

> Maybe we could do better here by having an example for which the comparer does something more interesting and realistic.

We could proceed with this PR (after addressing feedback) & iterate on a better sample in a follow up PR if you have any ideas.

Reply:
I'm not sure I'd add this. It's code that wouldn't be used in the sample, so might confuse the reader into thinking that it's somehow needed. Am I missing something?